e2e/test/scenarios/filters/reproductions/31340-long-column-name-search-results.cy.spec.js

```diff
@@ -0,0 +1,86 @@
+import {
+  assertDescendantNotOverflowsContainer,
+  assertIsEllipsified,
+  getBrokenUpTextMatcher,
+  popover,
+  restore,
+} from "e2e/support/helpers";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import { SAMPLE_DB_ID, SAMPLE_DB_SCHEMA_ID } from "e2e/support/cypress_data";
+
+const { PEOPLE_ID } = SAMPLE_DATABASE;
+
+const LONG_COLUMN_NAME =
+  "Some very very very very long column name that should have a line break";
+describe("issue 31340", function () {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.intercept("PUT", "/api/field/*").as("fieldUpdate");
+    cy.intercept("GET", "/api/field/*/search/*").as("search");
+
+    cy.visit(
+      `/admin/datamodel/database/${SAMPLE_DB_ID}/schema/${SAMPLE_DB_SCHEMA_ID}/table/${PEOPLE_ID}`,
+    );
+
+    cy.findByTestId("column-PASSWORD")
+      .findByDisplayValue("Password")
+      .type(`{selectAll}${LONG_COLUMN_NAME}`)
+      .blur();
+
+    cy.wait("@fieldUpdate");
+
+    cy.createQuestion(
+      {
+        query: {
+          "source-table": PEOPLE_ID,
+          limit: 2,
+        },
+      },
+      { visitQuestion: true },
+    );
+  });
+
+  it("should properly display long column names in filter options search results (metabase#31340)", () => {
+    cy.findAllByTestId("header-cell").contains(LONG_COLUMN_NAME).click();
+
+    popover().within(() => {
+      cy.findByText("Filter by this column").click();
+
+      cy.findByPlaceholderText(`Search by ${LONG_COLUMN_NAME}`).type(
+        "nonexistingvalue",
+      );
+
+      cy.wait("@search");
+
+      cy.findByText(
+        getBrokenUpTextMatcher(`No matching ${LONG_COLUMN_NAME} found.`),
+      )
+        .should("be.visible")
+        .then($container => {
+          cy.findByText(LONG_COLUMN_NAME).then($columnTextEl => {
+            // check that text block is not wider than the popover
+            assertDescendantNotOverflowsContainer(
+              $container[0],
+              $container.parent()[0],
+              "search results message",
+            );
+
+            // check that column name is within the text block
+            assertDescendantNotOverflowsContainer(
+              $columnTextEl[0],
+              $container[0],
+              "search results message",
+            );
+            // and it takes no more than 1 line
+            expect($columnTextEl[0].getBoundingClientRect().height).to.be.lte(
+              20,
+            ); // reasonable font size
```

Comment:
1. We don't need to hardcode the `20`
2. It depends on `line-height` in this case, but in general case also `padding`  and `border-width`.
3. This would make for a nice helper, e.g. `assertSingleLineOfText(element)`


```suggestion
            const lineHeight = parseFloat(
              window.getComputedStyle($columnTextEl[0]).lineHeight
            );

            expect($columnTextEl[0].getBoundingClientRect().height).to.be.lte(
              lineHeight
            );
```